### DISCUSSION TITLE
Receiving warning when trying to throw error from inside queryFn

### DISCUSSION THREAD

**Discussion by v-lax** [reduxjs/redux-toolkit#2110] [Q&A]
Hello! 

I have recently had to use queryFn to create a custom query that handles multiple async requests. I followed the example provided [here](https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#performing-multiple-requests-with-a-single-query). 

When implementing I noticed that when we throw an error (if the first async requests fails) then we get the following warning -> `An Unhandled error occured processing a request for the endpoint... In the case of an unhandled error, no tags will be provided` 

The warning went away after we returned an object with the error property. I noticed that [the documentation](https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#implementing-a-queryfn) for the queryFn function also says that it should return an object with either the `data` or `error` property. 

What I was confused about was why the example provided for preforming multiple async requests with a single query throws an error Rather than returning an object with an error property (like the documentation for queryFn says to)?

**Comment by phryneas** [comment]
Good point, that example needs an update. Could you maybe file a PR? :)

**Reply by v-lax** [reply]
Sure can!

**Reply by v-lax** [reply]
I'll try to have one up by the end of the day today eastern standard time. Just curious, do you have any specific guildelines around submitting a PR (I.e branching names, information a PR should have, creating an issue first before submitting the PR, ect)?

**Reply by phryneas** [reply]
Not really, we're happy about every PR :)

**Reply by v-lax** [reply]
Sounds great!
